srcjs/sidebar.js

```diff
@@ -5,17 +5,46 @@ if ($("section.sidebar").data("disable")) {
   $(".navbar > .sidebar-toggle").hide();
 }
 
-// Whenever the sidebar expand/collapse button is clicked:
-$(document).on("click", ".sidebar-toggle", function() {
+// Get the correct value for `input$sidebarCollapsed`, depending on
+// whether or not the left offset on the sidebar is negative (hidden
+// - so `input$sidebarCollapsed` should be TRUE) or 0 (shown - so
+// `input$sidebarCollapsed` should be FALSE). That we know of,
+// `$(".main-sidebar").is(":visible")` is always true, so there is
+// no need to check for that.
+var sidebarCollapsedValue = function() {
+  if ($(".main-sidebar").offset().left < 0) return(true);
+  else return(false);
+};
+
+// Whenever the sidebar changes from collapsed to expanded and vice versa,
+// call this function, so that we can trigger the resize event on the rest
+// of the window and also update the value for the sidebar's input binding.
+var sidebarChange = function() {
   // 1) Trigger the resize event (so images are responsive and resize)
   $(window).trigger("resize");
 
   // 2) Update the value for the sidebar's input binding
   var $obj = $('.main-sidebar.shiny-bound-input');
   var inputBinding = $obj.data('shiny-input-binding');
-  inputBinding.toggleValue($obj);
+  inputBinding.setValue($obj, sidebarCollapsedValue());
   $obj.trigger('change');
-});
+};
+
+// Whenever the sidebar finishes a transition (which it does every time it
+// changes from collapsed to expanded and vice versa), call sidebarChange()
+$( ".main-sidebar" ).on(
+  'webkitTransitionEnd otransitionend oTransitionEnd msTransitionEnd transitionend',
+    sidebarChange);
+
+// This fixes an edge case: when the app starts up with the sidebar already
+// collapsed (either because the screen is small or because
+// `dashboardSidebar(disable = TRUE`), make sure that `input$sidebarCollapsed`
+// is set to `FALSE`. Whenever this is the case, `$(".main-sidebar").offset().left`
+// is negative. That we know of, `$(".main-sidebar").is(":visible")` is always
+// true, so there is no need to check for that.
+if ($(".main-sidebar").offset().left < 0) {
+  $(".main-sidebar").attr("data-collapsed", "true");
```

Comment:
This is a little hard to follow since it sets the `data-collapsed` attribute directly, whereas in the `sidebarChange` function, it uses `inputBinding.setValue()`. Looking at the code, it's not obvious that they are both controlling the same thing.

Would it be possible to use the input binding here, or is this code executed too early for that to work?